### PR TITLE
Fix 1-4 scaling factors for both ligand and protein.

### PR DESCRIPTION
**Pull request by proteneer**
This PR addresses a long standing gotcha with our 1-4 scaling factors for electrostatics for both proteins and ligands. This PR implements them correctly (net effect is strengthening the 1-4 electrostatic interactions from 0.5 to 0.833).

### CODE REVIEW

File: timemachine/ff/handlers/openmm_deserializer.py [hunk 1]
```diff
@@ -111,7 +125,7 @@ def deserialize_nonbonded_force(force, N):
     beta = 2.0  # erfc correction
 
     # use the same scale factors for electrostatics and lj
```
Comment (maxentile) [inline timemachine/ff/handlers/openmm_deserializer.py:127]:
```suggestion
```

File: tests/test_topology.py [hunk 2]
```diff
@@ -12,9 +12,47 @@
...
+    for (i, j), qljw in zip(nb.idxs, nb_params):
+        kvs[(i, j)] = qljw
+
+    qs = ff.q_handle.parameterize(mol)
+    ljs = ff.lj_handle.parameterize(mol)
+
+    sigmas = ljs[:, 0]
+    epsilons = ljs[:, 1]
+
+    # 27-28 correspond to an H-O-C-H torsion, we expect:
```
Comment (jkausrelay) [inline tests/test_topology.py:43]: nit: label the atom idxs in H-O-C-H

File: timemachine/ff/handlers/openmm_deserializer.py [hunk 3]
```diff
@@ -85,9 +91,17 @@ def deserialize_nonbonded_force(force, N):
...
             lj_scale_factor = 1 - new_eps / expected_eps
 
-        scale_factors_.append(lj_scale_factor)
+        if expected_q == 0:
+            if new_q == 0:
+                q_scale_factor = 1
+            else:
+                raise RuntimeError("Divide by zero in charge calculation")
+        else:
+            q_scale_factor = 1 - new_q / expected_q
```
Comment (maxentile) [inline timemachine/ff/handlers/openmm_deserializer.py:100]: Don't 100% understand why a divide-by-zero is possible here. (But this works as-is: returns a solution to "(1 - scale_factor) * expected_q = new_q".)

**Review by maxentile — APPROVED** [review]
Presumably the bitwise tests will need to be updated, but otherwise looks good!

**Review by jkausrelay — APPROVED** [review]

File: tests/test_topology.py (hunk 2 above)
Comment (proteneer) [reply tests/test_topology.py:43]: willfix

File: timemachine/ff/handlers/openmm_deserializer.py (hunk 3 above)
Comment (proteneer) [reply timemachine/ff/handlers/openmm_deserializer.py:100]: Roughly the statement is saying: if the *desired* charge product is non-zero, but we're *starting* with a zero charge product, there's no number that we can multiply a zero starting charge with to arrive at a non-zero desired charge. The naming is confusing - I will rename it.

File: timemachine/ff/handlers/openmm_deserializer.py (hunk 1 above)
Comment (proteneer) [reply timemachine/ff/handlers/openmm_deserializer.py:127]: willfix

**Pull request merged**